docs/features/proto-generation.rst

```diff
@@ -1,11 +1,113 @@
+.. _proto-generation:
+
 Proto generation
 ================
+To be able to generate proto files you need to register your service first. 
+To do so please refer to `Service registration <https://socotecio.github.io/django-socio-grpc/server_and_service_register/#service-registration>`_ 
 
 Description
 -----------
+Proto files contain the classes, descriptors and controller logic (``pb2.py`` files) and proto message syntax (``.proto`` file) necessary to run a grpc server.
+In django-socio-grpc, proto files are generated from a ``ProtoSerializer`` class, 
+this serializer class should be declared as either the request or response of a ``grpc_action``, or as a ``serializer_class`` attribute inside a registered service.
+Fields contained in proto messages are mapped from the serializer fields.
+In order to generate these files and its contents, there is a django command to run whenever you add or modify your serializers (or whenever you register a new service that uses an existing serializer)
```

Comment:
it is dependent on grpcactions not serializers, serialiazers are one way among others to declare a request and response type